### ISSUE TITLE
(Feature request) Api for image storage and global options for  image copying in trilium

### ISSUE THREAD

**Issue by Onionbai** [TriliumNext/Notes#529]
### Api for image storage
For trilium, images can be stored inside or loaded from the web, but inside storage will surely increase the size of database. I know someone using image bed storing images before they are copied to trilium notes, but every time he would upload to the image bed first rather than directly paste it to trilium. But that is not convinent.
So I am thinking whether it is possible to open an api just for image storage (e.g. [easeimage2](https://github.com/icret/EasyImages2.0)), when we paste the image, it could be uploaded to the image bed. As users, they could choose the way images are stored, stored internally, externally or loaded from the web.
### global options for  image copying in trilium
In trilium, there are inconvinence like #521. Though surely we could press on shared button, then all the contents could be viewed and copied in browser, or we could export note into html, and then copy it, this is not convienent. 
So if it is possible to create an global button for image copying, options for internal link and image?

**Comment by perfectra1n** [comment]
If I'm understanding the request correctly, you would want a Trilium Note to load images from somewhere like [Imgur](https://imgur.com/) (after uploading the image to Imgur of course) and then placing the link to the image within Trilium?

**Comment by maphew** [comment]
@perfectra1n Sounds like wanting to be able to `Ctrl-V` paste images into Trilium and have Trilium take care of uploading to the 3rd party hosting service (presumably using a 3rd party API key) -- in other words to have it behave like Stack Overflow editing does.

I personally would not use such a feature, the risk of losing the external content far outweighs the cost of db storage size for me. I can see how others would choose differently though.

If such a feature were developed, I suggest making the default "3rd party" another Trilium db. This would allow differnet sync settings for each, and a "streaming vs local" experience similar to how Dropbox, Google Drive and Onedrive operate. The notes-db would be fully synchronized all the time, like now, and items in the assets-db would be on-demand. (Or configured to also be all-the-time.)

**Comment by openapphub** [comment]
It's OK to put them together. My personal attempt was to write a script to compress and convert the image attachments in the database and unify the webp format, and the effect was still OK. Current size 250 M, processed 170 M

**Comment by maphew** [comment]
> It's OK to put them together. My personal attempt was to write a script to compress and convert the image attachments in the database and unify the webp format, and the effect was still OK. Current size 250 M, processed 170 M

It's been my intention to open an enhanment issue to have webp as default formation and compression algorithim. And/or configurable to another format. Feel free to beat me to it. ;-)

**Comment by Onionbai** [comment]
> If I'm understanding the request correctly, you would want a Trilium Note to load images from somewhere like [Imgur](https://imgur.com/) (after uploading the image to Imgur of course) and then placing the link to the image within Trilium?

Sorry man, really busy these days. What I meamt is to separate the image storage from trilium database not simply to  load a image link which could be done by disable the function in  "Download images automatically for offline use." from somewhere just like imgur. Just like this, when you want to paster an image into trilium, inside trilium, if the api function enabled, the image will be uploaded to image bed, but as users, we feel nothing different.
![image](https://github.com/user-attachments/assets/78dbdb4b-4f00-45be-a2e8-4034bf2407ff)

However, as [maphew](https://github.com/maphew) mentioned,
> This would allow differnet sync settings for each, and a "streaming vs local" experience similar to how Dropbox, Google Drive and Onedrive operate. The notes-db would be fully synchronized all the time, like now, and items in the assets-db would be on-demand. (Or configured to also be all-the-time.)

this kind of option will bring sync problem which I didn't notice. And the risk of losing data after years is another big concern. I won't be happy too if someday in the future when I open one of the note, seeing all the images lost for the lose of function of image bed. That will be a catastrophy.

So it is better to keep what it is. And I saw some discussions, maybe I could try webp format to decrease the total database size (current 997M) someday,and thank you for your advice @openapphub, and I prefer this webp format as an option @maphew.

As for
>global options for image copying in trilium

I saw @eliandoran added #521  as a [bug], so... leave it to him😁.
